Makefile

```diff
@@ -47,6 +49,24 @@ build: bin $(CARGO_TARGET_DIR)
 	cargo build $(release)
 	cp $(CARGO_TARGET_DIR)/$(profile)/netavark bin/netavark$(if $(debug),.debug,)
 
+.PHONY: crate-publish
+crate-publish:
+	@if [ "$(publish-version)" = "" ]; then\
+		echo "publish-version not set. Usage example 'make crate-publish publish-version=v0.1.2'";\
+		echo "Make sure repo already contains tag for the version";\
+		exit 1;\
+	fi
+	@if [ "$(REMOTE_ORIGIN_URL)" != "https://github.com/containers/netavark" ]; then\
+		echo "You are working with a fork please use actual repo while publishing. Please clone https://github.com/containers/netavark";\
+		echo "Make sure repo already contains tag for $(publish-version)";\
+		exit 1;\
+	fi
+	git checkout tags/$(publish-version)
+	git branch
+	@echo "It is expected that you have already done 'cargo login' before running this command. If not command may fail later"
+	cargo publish --dry-run
+	cargo publish
+
```

Comment:
you never switch back to the previous git state

Reply:
done.